File: Java/Program/Membership.java
public class Membership { // Membership class
    // Attributes
    private String level;
    private double discount;
    // Constructors
    public Membership() {
        this("Basic", 0.0);
    }
    // Parameterized Constructor
    public Membership(String level, double discount) {
        this.level = level;
        this.discount = discount;
    }
    // Getters & Setters
    public String getLevel() { return level; }
    public void setLevel(String level) { this.level = level; }

    public double getDiscount() { return discount; }
    public void setDiscount(double discount) { this.discount = discount; }
    
    // Method to show membership details
    public void showMembership() {
        System.out.printf("%s\t%.2f%n", getLevel(), getDiscount());
    }
}

File: CPP/Program/Membership.cpp
#include <iostream>
#include <string>

using namespace std;

// This class represents a Membership
class Membership {
    // Attributes of the Membership
    protected:
        string level;
        double discount;

    public:
        // Default constructor
        Membership() {
            this->level = "Basic";
            this->discount = 0.0;
        }

        // Parameterized constructor
        Membership(string level, double discount) {
            this->level = level;
            this->discount = discount;
        }

        // Setters and Getters
        void setLevel(string level) {
            this->level = level;
        }

        string getLevel() {
            return this->level;
        }

        void setDiscount(double discount) {
            this->discount = discount;
        }

        double getDiscount() {
            return this->discount;
        }

        ~Membership() {
        }

        // For printing membership details
        void showMembership() {
            cout << getLevel() << "\t" << getDiscount() << endl;
        }
};